src/stores/PreferenceStore.ts

```diff
@@ -1,42 +1,73 @@
+        "scaling": 0,
+        "colormap": "inferno",
+        "percentile": 99.9,
+        "astColor": 4
     }`;
+
+    private defaultSettings;
     @observable json;
 
     // user configurable settings
     getScaling = (): FrameScaling => {
-        return this.json.scaling;
+        const scaling = this.json.scaling;
+        return (typeof scaling === "number" && RenderConfigStore.SCALING_TYPES.has(scaling)) ? scaling : this.defaultSettings.scaling;
+    };
+
+    getColormap = (): string => {
+        const colormap = this.json.colormap;
+        return (typeof colormap === "string" && RenderConfigStore.COLOR_MAPS_ALL.includes(colormap)) ? colormap : this.defaultSettings.colormap;
+    };
+
+    getPercentile = (): number => {
+        const percentile = this.json.percentile;
+        return (typeof percentile === "number" && RenderConfigStore.PERCENTILE_RANKS.includes(percentile)) ? percentile : this.defaultSettings.percentile;
+    };
+
+    getASTColor = (): number => {
+        const astColor = this.json.astColor;
+        return (typeof astColor === "number" && astColor >= 0 && astColor < AST.colors.length) ? astColor : this.defaultSettings.astColor;
     };
 
     @action setScaling = (newScaling: FrameScaling) => {
-        if (RenderConfigStore.SCALING_TYPES.has(newScaling)) {
-            this.json.scaling = newScaling;
-            localStorage.setItem("CARTA_preference", JSON.stringify(this.json));
-        }
+        this.json.scaling = RenderConfigStore.SCALING_TYPES.has(newScaling) ? newScaling : this.defaultSettings.scaling;
+        localStorage.setItem(PreferenceStore.CARTA_PREFERENCE, JSON.stringify(this.json));
     };
 
-    // for preference UI
-    @observable perferenceSelectedTab = "renderConfig";
-    @action setPreferenceSelectedTab(tabId: string) {
-        this.perferenceSelectedTab = tabId;
-    }
+    @action setColormap = (newColormap: string) => {
+        this.json.colormap = RenderConfigStore.COLOR_MAPS_ALL.includes(newColormap) ? newColormap : this.defaultSettings.colormap;
+        localStorage.setItem(PreferenceStore.CARTA_PREFERENCE, JSON.stringify(this.json));
+    };
+
+    @action setPercentile = (newPercentile: string) => {
+        const percentile = parseFloat(newPercentile);
+        this.json.percentile = (percentile && RenderConfigStore.PERCENTILE_RANKS.includes(percentile)) ? percentile : this.defaultSettings.percentile;
+        localStorage.setItem(PreferenceStore.CARTA_PREFERENCE, JSON.stringify(this.json));
+    };
+
+    @action setASTColor = (newColor: number) => {
+        this.json.astColor = (newColor >= 0 && newColor < AST.colors.length) ? newColor : this.defaultSettings.astColor;
+        localStorage.setItem(PreferenceStore.CARTA_PREFERENCE, JSON.stringify(this.json));
+    };
 
     constructor() {
-        const preference = localStorage.getItem("CARTA_preference");
+        this.defaultSettings = JSON.parse(PreferenceStore.DEFAULT_SETTINGS_JSON);
```

Comment:
why not just construct a default settings object directly, rather than using a default settings JSON and parsing it?